docs/VMR-Feature-Band-Support-Work-Plan.md

```diff
@@ -0,0 +1,92 @@
+
+Document convention:
+* There can be a variable number of feature band releases (e.g. 10.0.1xx, 10.0.2xx, 10.0.3xx, etc). For purposes of this document, we mainly care about the distinction between the 1xx branch and all the other non-1xx feature band branches. For simplicity, 2xx will be used as a notation to mean any non-1xx feature band branch.
+
+## Plan Details
+
+### Branch Management
+
+The 2xx branch sources should accurately reflect what gets produced in the output. Since the shared component are only being produced in the 1xx branch, those associated repos will be excluded from the 2xx branch upon creation.
+
+There should be tooling to help with this exclusion process for the VMR maintainers. Similarly, the associated projects in the `repo-projects` directory should be deleted along with updating the appropriate `RepositoryReference` usages.
+
+### Dependency Flow
+
+Repos in the 2xx branch of the VMR which have dependencies on repos solely defined in the 1xx branch (e.g. `sdk` repo's dependency on `runtime`) will need to have dependency flow that reflects the version of the package produced from the VMR's 1xx branch.
+
+```mermaid
+sequenceDiagram
+    autonumber
+
+    participant runtime as dotnet/runtime<br />release/10.0
+    participant VMR_1xx as VMR<br />release/10.0.1xx
+    participant VMR_2xx as VMR<br />release/10.0.2xx
+    participant VMR_3xx as VMR<br />release/10.0.3xx
+
+    runtime->>VMR_1xx: Flow of runtime change
+    Note over VMR_1xx: 📦 VMR build output packages are built
+    VMR_1xx->>VMR_2xx: Dependency flow <br />(runtime build output packages)
+    VMR_1xx->>VMR_3xx: Dependency flow <br />(runtime build output packages)
+```
+
+This will require changes to Maestro to allow output of the VMR to flow to another branch of the VMR.
+
+### Build
+
+* In addition to NuGet package, we also need to provide a way to have access to the runtime and ASP.NET Core artifacts (e.g. `dotnet-runtime-<version>-linux-x64.tar.gz`) from 2xx branches. This is necessary for the SDK to be able to bundle things together. This should be able to be done using https://ci.dot.net, just as it is done in product repo builds.
+* Since Arcade repo will not be available, all repos will need to use the bootstrap Arcade (`UseBootstrapArcade=true`).
+* While the `scenario-tests` repo doesn't have differing code for each feature band, it will still need to exist in each of the VMR's feature band branches because it is built as part of the pipeline run and used to validate the product. This is ok since it is not part of the shipped binaries.
+
+❓ Open Question:
+There will need to be some way to detect when we are in one of these 2xx branches. What mechanism should be used for that? Maybe checking `packageVersion` of the sdk in `source-manifest.json`?
```

Comment:
Where do we need this knowledge?

Reply:
Will there be any canonical way to determine the SDK version at the VMR root level?

/cc @premun

Reply:
This will have to be set somewhere in `Versions.props`